Cython/Compiler/Nodes.py

```diff
@@ -729,7 +729,8 @@ def analyse(self, return_type, env, nonempty=0, directive_locals=None, visibilit
                 # Use an explicit exception return value to speed up exception checks.
                 # Even if it is not declared, we can use the default exception value of the return type,
                 # unless the function is some kind of external function that we do not control.
-                if return_type.exception_value is not None and (visibility != 'extern' and not in_pxd):
+                if (return_type.exception_value is not None and (visibility != 'extern' and not in_pxd)
+                        and not isinstance(self.base, CPtrDeclaratorNode)):
```

Comment:
I don't hugely like this (it seems like it'd be better to handle it after the pointer type was analysed) but it's hard to see another way of doing it without a big reordering of the code

Reply:
Ok. Could you at least add a comment that this is more of a hack and should be moved elsewhere eventually?

Reply:
Done